api/apps/api/src/modules/access-control/scenarios-acl/scenario-access-control.ts

```diff
@@ -11,4 +19,28 @@ export abstract class ScenarioAccessControl {
     projectId: string,
   ): Promise<Permit>;
   abstract canCloneScenario(userId: string, projectId: string): Promise<Permit>;
+  abstract canReleaseLock(userId: string, projectId: string): Promise<Permit>;
+  abstract acquireLock(
+    userId: string,
+    scenarioId: string,
+  ): Promise<Either<typeof forbiddenError | AcquireFailure, ScenarioLockDto>>;
+  abstract releaseLock(
+    userId: string,
+    scenarioId: string,
+    projectId: string,
+  ): Promise<
+    Either<
+      typeof forbiddenError | typeof lockedByAnotherUser | typeof noLockInPlace,
+      void
+    >
+  >;
+  abstract canEditScenarioAndLockLogicIsCorrect(
```

Comment:
I think I left this name because previously I was trying to attempt including the permissions of a Project Owner/Contributor in this function and I didn't rename it when I opted for leaving that out because it was getting quite confusing, but I think yeah your name is a better suggestion then.